src/main/java/io/cdap/plugin/successfactors/common/util/AccessToken.java

```diff
@@ -0,0 +1,362 @@
+
+      // Support sha256 signing algorithm for external oauth saml assertion
+      signature.setSignatureAlgorithm(SignatureConstants.ALGO_ID_SIGNATURE_RSA_SHA256);
+
+      assertion.setSignature(signature);
+      Configuration.getMarshallerFactory().getMarshaller(assertion).marshall(assertion);
+      Signer.signObject(signature);
+    } catch (Exception e) {
+      throw new Exception("Failure in signing the SAML2 assertion", e);
+    }
+    return assertion;
+  }
+
+  private static PrivateKey generatePrivateKey(String privateKeyString) {
+    KeyFactory keyFactory = null;
+    try {
+      keyFactory = KeyFactory.getInstance("RSA");
+    } catch (NoSuchAlgorithmException e) {
+      throw new RuntimeException(e.getMessage(), e);
+    }
+    String pk2 = privateKeyString;
+    try {
+      pk2 = new String(Base64.decodeBase64(pk2), "UTF-8");
+    } catch (UnsupportedEncodingException e) {
+      throw new RuntimeException(e.getMessage(), e);
+    }
+    String[] strs = pk2.split("###");
+    if (null != strs && strs.length == 2) {
+      privateKeyString = strs[0];
+    }
+    PKCS8EncodedKeySpec privateKeySpec = new PKCS8EncodedKeySpec(Base64.decodeBase64(privateKeyString));
+    PrivateKey privateKey = null;
+    try {
+      privateKey = keyFactory.generatePrivate(privateKeySpec);
+    } catch (InvalidKeySpecException e) {
+      throw new RuntimeException(e.getMessage(), e);
+    }
+    return privateKey;
+  }
+
+  public String getAssertionToken() throws Exception {
+
+    /**
+     * Below code is to produce signed assertion via code directly using provided
+     * input
+     */
+    String tokenUrl = config.getTokenURL(), clientId = config.getClientId(),
+      privateKey = config.getPrivateKey(), userId = config.getUserId();
+    boolean useUserNameAsUserId = false;
+    int expireInMinutes = 10;
+
+    try {
+
+      if (tokenUrl != null && clientId != null && privateKey != null && userId != null) {
+        LOG.info("All properties are set, generating the SAML Assertion...");
+
+        String signedSAMLAssertion = generateSignedSAMLAssertion(clientId, userId, tokenUrl, privateKey,
+          expireInMinutes, useUserNameAsUserId);
+        LOG.info("The generated Signed SAML Assertion is:");
+        return signedSAMLAssertion;
+      }
+    } catch (Exception e) {
+      throw new Exception("Fail to generate SAML Assertion due to " + e.getMessage());
+    }
+    return null;
+  }
+
+  public String getAccessToken(String assertionToken) throws IOException {
+    HttpClient client = HttpClientBuilder.create().build();
+
+    // Build POST request
+    HttpPost request = new HttpPost(URI.create("https://apisalesdemo2.successfactors.eu/oauth/token"));
+
+    // Set headers
+    request.setHeader("Authorization", "none");
+    request.setHeader("Content-Type", "application/x-www-form-urlencoded");
+
+    // Build request body
+    StringBuilder body = new StringBuilder();
+    body.append("client_id" + "=" + config.getClientId());
```

Comment:
Using `StringBuilder` and still making use of `+` is like under utilisation.
```suggestion
    body.append("client_id=").append(config.getClientId());
```

Reply:
done